modin/pandas/groupby.py

```diff
@@ -690,10 +840,20 @@ def hist(self):
         return self._default_to_pandas(lambda df: df.hist())
 
     def quantile(self, q=0.5, **kwargs):
+        selection = self._non_by_selection
+
+        dtypes = self._df.dtypes if selection is None else self._df.dtypes[selection]
+        if not isinstance(dtypes, pandas.Series):
+            dtypes = pandas.Series(dtypes)
+        if dtypes.map(lambda x: x == np.dtype("O")).all():
+            raise TypeError("'quantile' cannot be performed against 'object' dtypes!")
```

Comment:
previously this was done naturally at the backend, but didn't work with some corner cases, added this explicit check to the front-end so now users can get proper errors as soon as possible